src/reducers/AuthReducer.js

```diff
@@ -24,16 +47,46 @@ export default (state = INITIAL_STATE, action) => {
             return ({...state, appReady: true});
 
         case GUEST_TOKEN_SET:
-            return ({...state, token: payload.token, appReady: payload.appReady});
+            const guestUser = {name: payload.token};
+            return ({...state, token: payload.token, currentUser: guestUser, appReady: payload.appReady});
 
         case FACEBOOK_LOGIN_SUCCESS:
-            return ({...state, token: payload.token, appReady: payload.appReady});
+            return ({...state, token: payload.token, currentUser: payload.currentUser, appReady: payload.appReady});
```

Comment:
Now passing around the `currentUser` along with the `token`. Again, we'll probably be able to remove `token` very soon. Probably in my next PR.